spacelift/type_conversions.go

```diff
@@ -88,3 +88,19 @@ func toOptionalStringMap(input interface{}) *structs.StringMap {
 	}
 	return &customHeaders
 }
+
+// getOptionalBool checks if user has explicitly set a boolean value in the configuration and returns it as a pointer to graphql.Boolean.
+// If the value is not set in the configuration, it returns nil.
+func getOptionalBool(d *schema.ResourceData, key string) *graphql.Boolean {
```

Comment:
Could we provide some tests?

Reply:
I tried to make some unit tests for this function and failed. `*schema.ResourceData` contains only private fields and test util function provided by terraform to generate `*schema.ResourceData` does not populate fields that `GetRawConfig` is using.

Technically this function has some coverage, because its invoked by resources that are tested